src/Symfony/Component/JsonPath/JsonCrawler.php

```diff
@@ -562,16 +824,260 @@ private function evaluateRecursive(mixed $value): array
+        ) {
+            return true;
+        }
+
+        if ($leftIsNothing || $rightIsNothing) {
+            return false;
+        }
+
+        if ((\is_int($left) || \is_float($left)) && (\is_int($right) || \is_float($right))) {
+            return $left == $right;
+        }
+
+        if (\is_string($left) && \is_string($right) || \is_bool($left) && \is_bool($right)) {
+            return $left === $right;
+        }
+
+        if (null === $left && null === $right) {
+            return true;
+        }
+
+        // arrays must have equal length and equal corresponding elements
+        if (\is_array($left) && \is_array($right)) {
+            return $this->compareArraysDeep($left, $right);
+        }
+
+        // objects must have identical names and equal corresponding values
+        if ($left instanceof \stdClass && $right instanceof \stdClass) {
+            return $this->compareObjectsDeep($left, $right);
+        }
+
+        // null (missing property) equals 0 when compared to function results
+        if (null === $left && 0 === $right || 0 === $left && null === $right) {
+            return true;
+        }
+
+        // different types are not equal
+        return false;
+    }
+
+    private function compareArraysDeep(array $left, array $right): bool
+    {
+        $leftIsList = array_is_list($left);
+        $rightIsList = array_is_list($right);
+        $leftCount = \count($left);
+
+        if ($leftIsList !== $rightIsList || $leftCount !== \count($right)) {
+            return false;
+        }
+
+        if ($leftIsList) {
+            for ($i = 0; $i < $leftCount; ++$i) {
+                if (!$this->compareEquality($left[$i], $right[$i])) {
+                    return false;
+                }
+            }
+        } else {
+            foreach ($left as $key => $value) {
+                if (!\array_key_exists($key, $right) || !$this->compareEquality($value, $right[$key])) {
+                    return false;
+                }
+            }
+        }
+
+        return true;
+    }
+
+    private function compareObjectsDeep(\stdClass $left, \stdClass $right): bool
+    {
+        $leftVars = get_object_vars($left);
+        $rightVars = get_object_vars($right);
+
+        if (\count($leftVars) !== \count($rightVars)) {
+            return false;
+        }
+
+        foreach ($leftVars as $key => $value) {
+            if (!property_exists($right, $key) || !$this->compareEquality($value, $rightVars[$key])) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private function compareOrdering(mixed $left, mixed $right, string $operator): bool
+    {
+        if (null === $left || null === $right) {
+            return match ($operator) {
+                '>=', '<=' => $left === $right,
+                default => false,
+            };
+        }
+
+        if ((\is_int($left) || \is_float($left)) && (\is_int($right) || \is_float($right))) {
```

Comment:
the case for bool below can be remove after this change:
```suggestion
        if ((\is_int($left) || \is_float($left)) && (\is_int($right) || \is_float($right)) || \is_bool($left) && \is_bool($right)) {
```